library/Booster/Syntax/Json/Internalise.hs

```diff
@@ -130,10 +160,14 @@ internaliseTermRaw qq allowAlias sortVars definition@KoreDefinition{sorts, symbo
                 from' <- lookupInternalSort' from
                 to' <- lookupInternalSort' to
                 arg' <- recursion arg
+                unless (not (coerce checkSubsorts) || from' `isSubsort` to') $
```

Comment:
We have to disable this check when internalising the kore definition, otherwise we would have to re-structure the module internalisation into two passes, one where we only internalise sorts and sub-sorts and then the second pass internalises all other axioms. Depends on how much we trust the front-end to emit correct injections :P

Given that we now have three flags, I decided to newtype them to avoid stupid mix-ups when passing the parameters around.